packages/langium/test/grammar/references/grammar-scoping.test.ts

```diff
@@ -0,0 +1,60 @@
+/******************************************************************************
+ * Copyright 2022 TypeFox GmbH
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License, which is available in the project root.
+ ******************************************************************************/
+
+import { Utils } from 'vscode-uri';
+import { createLangiumGrammarServices, DefaultIndexManager, DefaultLangiumDocuments, EmptyFileSystem, Grammar } from '../../../src';
+import { CrossReference, InferredType } from '../../../src/grammar/generated/ast';
+import { parseHelper } from '../../../src/test';
+
+describe('Type Linking', () => {
+
+    const services = createLangiumGrammarServices(EmptyFileSystem);
+    const locator = services.grammar.workspace.AstNodeLocator;
+    const parse = parseHelper<Grammar>(services.grammar);
+
+    beforeAll(() => {
+        (services.shared.workspace.IndexManager as DefaultIndexManager).reset();
+        (services.shared.workspace.LangiumDocuments as DefaultLangiumDocuments).clear();
+    });
+
+    test('Linking in same file', async () => {
+        const grammar = await parse(`
+        grammar Debug
+        entry Model:
+            (b+=B | a+=A)*;
+        A infers A:
+            'A' a=ID;
+        B infers B:
+            'B' b=[A:ID];
+        terminal ID: /[_a-zA-Z][\\w_]*/;
+        `);
+        await services.shared.workspace.DocumentBuilder.build([grammar]);
+        const reference = locator.getAstNode<CrossReference>(grammar, 'rules@2/definition/elements@1/terminal');
+        const inferredType = locator.getAstNode<InferredType>(grammar, 'rules@1/inferredType');
+        expect(reference?.type.ref).toBe(inferredType);
+    });
+
+    test('Linking in other file', async () => {
```

Comment:
Do you mind adding another test here, which tests that cross references to type declarations (interfaces) are correctly resolved even in the face of a parser rule with the same name, which returns this interface? Something like this:

```
interface X {
  ...
}
X returns X: ...;

// In a separate file
Y: ref=[X:ID];
```